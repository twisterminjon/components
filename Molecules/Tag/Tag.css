.tag {
  display: inline-block;

  border: none;
  border-radius: 6px;

  text-transform: uppercase;

  background-color: var(--brandcolor_dark);
  color: var(--white);

  font-size: 14px;
  line-height: 30px;

  overflow: hidden;
}

.tag-text {
  display: flex;
  margin: 0 6px 0 8px;
  word-break: break-word;
}

.tag--inactive {
  color: var(--button_color_disabled);
}

.tag--fluid {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.tag-button {
  height: 30px;
  width: 30px;
}
