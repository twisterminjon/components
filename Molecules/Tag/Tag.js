import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import DeleteButtonTiny from '../../Atoms/DeleteButtonTiny/DeleteButtonTiny'
import './Tag.css'

/**
 * <Tag id={id} onAction={(id, name) => onTagDelete(id, name)} etc... >
 *    <Tag.Icon><IconEnvelope color={contentColor()} /></Tag.Icon>
 *    <Tag.Label>{name}</Tag.label>
 *     <Tag.Button>
 *      <DeleteButtonTiny  data-testid={`patientsurvey-delete-${name}`} />
 *     </Tag.button>
 * </Tag>
 */
export default class Tag extends Component {
  static propTypes = {
    /** Text to display in the tag */
    label: PropTypes.string.isRequired,
    /** id of the item being displayed */
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    /** If true, will expand to the width of the container */
    fluid: PropTypes.bool,
    /** A tag can display an inactive state */
    inActive: PropTypes.bool,
    /** Called when the delete button is clicked */
    onDelete: PropTypes.func.isRequired,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
    /** Optional view  */
    ContentView: PropTypes.any, // FIXME: Define better type
  }
  static defaultProps = {
    fluid: false,
    inActive: false,
    className: '',
    style: {},
    ContentView: ({ id, label, onDelete }) => (
      <Fragment>
        <span className="tag-text" data-testid={`tag-${label}-text`}>
          {label}
        </span>
        <DeleteButtonTiny
          onClick={() => onDelete(id, label)}
          data-testid={`tag-${label}-delete-button`}
          className={`tag-button`}
        />
      </Fragment>
    ),
  }

  render() {
    const {
      label,
      onDelete,
      className,
      fluid,
      inActive, // FIXME: Rename this prop to "inactive"
      style,
      id,
      ContentView,
    } = this.props

    const fluidClass = fluid ? 'tag--fluid' : ''
    const inActiveClass = inActive ? 'tag--inactive' : ''

    const dataTestId = this.props['data-testid'] ? this.props['data-testid'] : `tag-${label}`

    return (
      <div className={`tag ${inActiveClass} ${fluidClass} ${className}`} style={style} data-testid={dataTestId}>
        <ContentView id={id} label={label} onDelete={onDelete} inactive={inActive} />
      </div>
    )
  }
}
