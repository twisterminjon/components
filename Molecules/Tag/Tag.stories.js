import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import centered from '@storybook/addon-centered/react'

import Tag from './Tag'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('Tag', () => {
  const labelLabel = 'label'
  const labelDefaultValue = 'D-list super heros'
  const labelValue = text(labelLabel, labelDefaultValue)

  const inActiveLabel = 'inActive'
  const inActiveDefaultValue = false
  const inActiveValue = boolean(inActiveLabel, inActiveDefaultValue)

  return (
    <Tag
      id="1"
      label={labelValue}
      inActive={inActiveValue}
      onDelete={() => {
        alert('delete clicked')
      }}
    />
  )
})
