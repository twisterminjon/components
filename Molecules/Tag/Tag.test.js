import React from 'react'
import Tag from './Tag'

describe('Tag', () => {
  const label = 'Legit'
  const id = '1'
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<Tag id={id} label={label} onDelete={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('displays the correct text', () => {
    // Not a shallow render since the internal tag view is a sub-rendered component
    const wrapper = window.mount(<Tag id={id} label={label} onDelete={mockFun} />)

    expect(wrapper.find('[data-testid="tag-Legit-text"]').text()).toEqual(label)
  })

  it('calls the delete function', () => {
    const wrapper = window.mount(<Tag id={id} label={label} onDelete={mockFun} />)

    wrapper
      .find('[data-testid="tag-Legit-delete-button"]')
      .last() // Use last rendered component as the virtual DOM shows React components in addition to actual DOM components
      .simulate('click')
    expect(mockFun).toHaveBeenCalled()
  })
})
