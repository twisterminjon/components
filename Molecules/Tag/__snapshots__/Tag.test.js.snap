// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches the snapshot 1`] = `
<div
  className="tag   "
  data-testid="tag-Legit"
  style={Object {}}
>
  <ContentView
    id="1"
    inactive={false}
    label="Legit"
    onDelete={[MockFunction]}
  />
</div>
`;
